lib/EarlyFunctionAnalysis/FunctionSummaryOracle.cpp

```diff
@@ -25,13 +25,18 @@ struct PrototypeImporter {
                             const model::TypePath &Prototype) {
     FunctionSummary Summary(Attributes, ABICSVs, {}, {}, 0);
 
-    std::set<llvm::GlobalVariable *> PreservedRegisters;
-    for (Register R : abi::FunctionType::calleeSavedRegisters(Prototype)) {
-      llvm::StringRef Name = model::Register::getCSVName(R);
-      if (llvm::GlobalVariable *CSV = M.getGlobalVariable(Name, true))
-        PreservedRegisters.insert(CSV);
-    }
-
+    // Drop known to be preserved registers from `Summary.ClobberedRegisters`.
+    auto RegisterToCSV = [this](Register Register) -> llvm::GlobalVariable * {
+      return M.getGlobalVariable(model::Register::getCSVName(Register), true);
+    };
+    auto NullptrFilter = [](const auto *Pointer) -> bool {
+      return Pointer != nullptr;
+    };
+
+    auto PreservedRegisters = abi::FunctionType::calleeSavedRegisters(Prototype)
+                              | std::views::transform(RegisterToCSV)
+                              | std::views::filter(NullptrFilter)
+                              | revng::to<std::set<llvm::GlobalVariable *>>();
```

Comment:
`:O`

What do you think of:

```c++
    auto ToCSV = std::views::transform([this](Register Register) -> llvm::GlobalVariable * {
      return M.getGlobalVariable(model::Register::getCSVName(Register), true);
    });
    auto RemoveNullptr = std::views::filter([](const auto *Pointer) -> bool {
      return Pointer != nullptr;
    });

    auto PreservedRegisters = abi::FunctionType::calleeSavedRegisters(Prototype)
                              | ToCSV
                              | RemoveNullptr
                              | revng::to<std::set<llvm::GlobalVariable *>>();
```

I think this kind of paradigm is very effective when you first prepare all the components with very explicative, obvious, high level names and then you combine them without noise.

Reply:
Done.

(I went with `TransformToCSV` and `IgnoreNullptr`)

the only problem is that I don't really like how clang-format reformatted it into
```cpp
    auto Preserved = abi::FunctionType::calleeSavedRegisters(Prototype)
                     | TransformToCSV | IgnoreNullptr
                     | revng::to<std::set<llvm::GlobalVariable *>>();
```
But I guess I can live with that.